libraries/chain/include/sysio/chain/authority.hpp

```diff
@@ -133,6 +139,10 @@ struct shared_key_weight {
          [&]<class T>(const T& k1r1) {
             key.pubkey.emplace<T>(k1r1);
          },
+         [&](const fc::crypto::bls::public_key_shim& blsa) {
+            // TODO: @heifner, not sure how this is used? Is it needed for BLS support?
+            //key.pubkey.emplace<fc::crypto::bls::public_key_shim>(blsa.unwrapped());
+         },
```

Comment:
Yes, this is used if we are going to support allowing BLS keys in authorities. If we are not going to allow BLS keys in authorities they we should SYS_ASSERT that so they can't be stored in chainbase.

I think you should be able to just remove this and let it default to the `k1r1` `class T` case. We should rename the `k1r1` to `shim` as it is more than just k1 r1 types now.